src/Microsoft.DotNet.Build.Tasks/PrereleaseResolveNuGetPackageAssets.cs

```diff
@@ -669,22 +670,239 @@ private static string TryGetRuntimeVersion(string path)
             uint actualBufferSize = 0;
             int bufferLength = 11; // 11 is the length of a runtime version and null terminator v2.0.50727/0
 
-            do
+            try
             {
-                runtimeVersion = new StringBuilder(bufferLength);
-                hresult = NativeMethods.GetFileVersion(path, runtimeVersion, bufferLength, out actualBufferSize);
-                bufferLength = bufferLength * 2;
+                do
+                {
+                    runtimeVersion = new StringBuilder(bufferLength);
+                    hresult = NativeMethods.GetFileVersion(path, runtimeVersion, bufferLength, out actualBufferSize);
```

Comment:
Was just trying to follow what MSBuild did more or less (they actually switch on the platform using the legacy Environment.OSVersion stuff).  Can change to just use the PE Reader.